README.md

```diff
@@ -9,6 +9,142 @@ production code.
 
 See the source for detailed documentation.
 
+Sticky Loader
+-------------
+
+A sticky loader is a thin wrapper around `taskcluster-lib-loader` to support
+dependency injection. It "remembers" each value it has returned and will return
+it again on the next call; it can also have a dependency injected.  Use it like
+this in `helper.js`:
+
+```javascript
+const {stickyLoader} = require('taskcluster-lib-testing');
+const load = require('../src/server');
+
+exports.load = stickyLoader(load);
+
+setup(function() {
+  exports.load.reset();
+  exports.load.inject('profile', 'test');
+  exports.load.inject('process', 'test');
+});
+```
+
+then, in test scripts:
+
+```javascript
+const {load} = require('./helper');
+
+setup(async function() {
+  load.inject('cfg', { /* fake config */});
+  const SomeTable = await load('SomeTable');
+  await SomeTable.create({ /* ... */ });
+});
+
+test(async function() {
+  const component = await load('some-component');
+  // component will be loaded with the fake config and with
+  // the same instance of SomeTable that we set up above
+});
+```
+
+Secrets
+-------
+
+This class handles getting secrets for tests, and easily determining what
+secrets are available.  It integrates with `typed-env-config`.  Set it up by
+in `test/helper.js`:
+
+```javascript
+const {Secrets} = require('taskcluster-lib-testing');
+
+exports.secrets = new Secrets({
+  secretName: 'project/taskcluster/testing/taskcluster-lib-foo',
+  secrets: {
+   pulse: [
+     {env: 'PULSE_USERNAME', cfg: 'pulse.username', name: 'username'},
```

Comment:
Maybe a bit right upfront about what `env`, `cfg`, and `name` mean in this context?